src/osdc/Objecter.cc

```diff
@@ -2761,14 +2761,8 @@ int Objecter::_calc_target(op_target_t *t, Connection *con, bool any_change)
   vector<int> up, acting;
   ps_t actual_ps = ceph_stable_mod(pgid.ps(), pg_num, pg_num_mask);
   pg_t actual_pgid(actual_ps, pgid.pool());
-  pg_mapping_t pg_mapping;
-  pg_mapping.epoch = osdmap->get_epoch();
-  if (lookup_pg_mapping(actual_pgid, &pg_mapping)) {
-    up = pg_mapping.up;
-    up_primary = pg_mapping.up_primary;
-    acting = pg_mapping.acting;
-    acting_primary = pg_mapping.acting_primary;
-  } else {
+  if (!lookup_pg_mapping(actual_pgid, osdmap->get_epoch(), &up, &up_primary,
+                         &acting, &acting_primary)) {
     osdmap->pg_to_up_acting_osds(actual_pgid, &up, &up_primary,
                                  &acting, &acting_primary);
     pg_mapping_t pg_mapping(osdmap->get_epoch(),
```

Comment:
Need to fix this constructor.  The copies it does don't show up in the profile because once the cache is populated it is called again only after the osdmap changes but let's not leave it on the table anyway.